packages/swapper/src/swappers/zrx/ZrxSwapper.ts

```diff
@@ -27,62 +29,75 @@ import { zrxBuildTrade } from './zrxBuildTrade/zrxBuildTrade'
 import { zrxExecuteTrade } from './zrxExecuteTrade/zrxExecuteTrade'
 
 export type ZrxSwapperDeps = {
-  adapter: ethereum.ChainAdapter
+  adapter: ethereum.ChainAdapter | avalanche.ChainAdapter
   web3: Web3
 }
 
-export class ZrxSwapper implements Swapper<'eip155:1'> {
+export type ZrxSupportedChainIds = KnownChainIds.EthereumMainnet | KnownChainIds.AvalancheMainnet
+
+export class ZrxSwapper<T extends ZrxSupportedChainIds> implements Swapper<ZrxSupportedChainIds> {
   public static swapperName = 'ZrxSwapper'
   deps: ZrxSwapperDeps
+  chainId: ChainId
 
   constructor(deps: ZrxSwapperDeps) {
     this.deps = deps
+    this.chainId = deps.adapter.getChainId()
   }
 
   // noop for zrx
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   async initialize() {}
 
   getType() {
-    return SwapperType.Zrx
+    switch (this.chainId) {
+      case KnownChainIds.EthereumMainnet:
+        return SwapperType.ZrxEthereum
+      case KnownChainIds.AvalancheMainnet:
+        return SwapperType.ZrxAvalanche
+      default:
+        throw new SwapError('[getType]', {
+          code: SwapErrorTypes.UNSUPPORTED_CHAIN
+        })
+    }
   }
 
   async buildTrade(args: BuildTradeInput): Promise<ZrxTrade> {
     return zrxBuildTrade(this.deps, args)
   }
 
-  async getTradeQuote(input: GetTradeQuoteInput): Promise<TradeQuote<'eip155:1'>> {
+  async getTradeQuote(input: GetTradeQuoteInput): Promise<TradeQuote<ZrxSupportedChainIds>> {
```

Comment:
If `getZrxTradeQuote` is updated to a generic that can pass the type through to the return type, we can use `T` here as well.

Reply:
Fixed generics in https://github.com/shapeshift/lib/pull/850/commits/bef008774c30162f801d4d8edffe49da7289390c, though I don't know why I had to cast - I couldn't get it to infer.